app/src/main/java/com/taylorgirard/comicconvo/adapters/MessageAdapter.java

```diff
@@ -0,0 +1,149 @@
+package com.taylorgirard.comicconvo.adapters;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.bumptech.glide.Glide;
+import com.bumptech.glide.load.resource.bitmap.CircleCrop;
+import com.parse.ParseException;
+import com.parse.ParseFile;
+import com.parse.ParseUser;
+import com.taylorgirard.comicconvo.R;
+import com.taylorgirard.comicconvo.models.Message;
+
+import java.util.List;
+
+public class MessageAdapter extends RecyclerView.Adapter<MessageAdapter.MessageViewHolder> {
+
+    private static final int MESSAGE_OUTGOING = 123;
+    private static final int MESSAGE_INCOMING = 321;
+
+    private List<Message> messages;
+    private Context context;
+    private ParseUser match;
+    private ParseUser user = ParseUser.getCurrentUser();
+
+    public MessageAdapter(Context context, ParseUser match, List<Message> messages){
+        this.messages = messages;
+        this.match = match;
+        this.context = context;
+    }
+
+    @Override
+    public int getItemCount() {
+        return messages.size();
+    }
+
+    @NonNull
+    @Override
+    public MessageViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        Context context = parent.getContext();
+        LayoutInflater inflater = LayoutInflater.from(context);
+
+        if (viewType == MESSAGE_INCOMING){
+            View contactView = inflater.inflate(R.layout.message_incoming, parent, false);
+            return new IncomingMessageViewHolder(contactView);
+        } else if (viewType == MESSAGE_OUTGOING){
+            View contactView = inflater.inflate(R.layout.message_outgoing, parent, false);
+            return new OutgoingMessageViewHolder(contactView);
+        } else{
+            throw new IllegalArgumentException("Unknown view type");
+        }
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull MessageViewHolder holder, int position) {
+        Message message = messages.get(position);
+        holder.bindMessage(message);
+    }
+
+    @Override
+    public int getItemViewType(int position) {
+        if (isMe(position)){
+            return MESSAGE_OUTGOING;
+        } else {
+            return MESSAGE_INCOMING;
+        }
+    }
+
+    private boolean isMe(int position){
```

Comment:
how does this look compared to sending to users that are not yourself?

Reply:
This simply checks whether the sender of the message is you to pick the correct ViewHolder for the message. The message layout is different depending on if it is outgoing or incoming. You cannot send messages to yourself.